### PR TITLE
fix(arrayjoin) Allow multiple array join columns during parsing.

### PR DESCRIPTION
**Pull request by fpacifici**
https://github.com/getsentry/snuba/pull/1212/ enforces that only one arrayJoin column is present in a query. 
This is a problem with measurements where we stopped having a special syntax for arrayjoin (like tags_key and tags_value) and we require the arrayJoin to be explicit.

This supports the same type of check of PR #1212 but with multiple array join columns.

There are followups to make #1212 actually behave as expected:
- deal with aliases since at the point in the parsing where the check is done aliases have not been expanded yet.
- deal with tags_key and tags_Value that, at the time of the check are not expressed as arrayJoin
- deal with all clauses in the query that can contain an arrayJoin.

### CODE REVIEW

File: snuba/query/parser/__init__.py [hunk 1]
```diff
@@ -137,32 +138,34 @@ def build_selected_expressions(
...
             if isinstance(select_expr.expression, FunctionCall):
                 if select_expr.expression.function_name == "arrayJoin":
-                    if arrayjoin:
-                        raise ParsingException(
-                            "Only one arrayJoin(...) call is allowed in a query."
-                        )
-
                     parameters = select_expr.expression.parameters
                     if len(parameters) != 1 or not isinstance(parameters[0], Column):
                         raise ParsingException(
```
Comment (evanh) [inline snuba/query/parser/__init__.py:158]: If we allow multiple columns, then should this check go away too?

**Review by evanh — APPROVED** [review]
One question about multiple arguments to `arrayJoin(...)`.

File: snuba/query/parser/__init__.py (hunk 1 above)
Comment (fpacifici) [reply snuba/query/parser/__init__.py:158]: Not enough. This PR now depends on https://github.com/getsentry/snuba/pull/1334 which takes care of multiple columns array joined in the query.

Comment (fpacifici) [reply snuba/query/parser/__init__.py:158]: No, I think every single arrayJoin call should have at most one parameter.

Comment (fpacifici) [reply snuba/query/parser/__init__.py:158]: @evanh actually I made a change here since Tony mentioned they need array join of literals

**Review by evanh — APPROVED** [review]

**Pull request merged**